cpp/daal/include/services/internal/sycl/execution_context_sycl.h

```diff
@@ -161,36 +128,27 @@ class OpenClKernelFactory : public Base, public ClKernelFactoryIface
             const bool isOpenCLBackendAvailable = !_deviceQueue.get_device().template get_info<cl::sycl::info::device::opencl_c_version>().empty();
             if (isOpenCLBackendAvailable)
             {
-        #endif // DAAL_DISABLE_LEVEL_ZERO \
-            // OpenCl branch
-                auto kernelRef = OpenClKernelRef(_currentProgramRef->get(), kernelName, &localStatus);
-                if (!localStatus.ok())
-                {
-                    services::internal::tryAssignStatus(status, localStatus);
-                    return KernelPtr();
-                }
+        #endif // DAAL_DISABLE_LEVEL_ZERO
+
+                // OpenCL branch
+                auto kernelRef = OpenClKernelRef(_currentProgramRef->get(), kernelName, status);
+                DAAL_CHECK_STATUS_RETURN_IF_FAIL(status, KernelPtr());
+
                 kernel.reset(new OpenClKernelNative(_executionTarget, *_currentProgramRef, kernelRef));
         #ifndef DAAL_DISABLE_LEVEL_ZERO
             }
             else
             {
                 // Level zero branch
-                auto kernelRef = OpenClKernelLevelZeroRef(kernelName, &localStatus);
+                auto kernelRef = OpenClKernelLevelZeroRef(kernelName, status);
+                DAAL_CHECK_STATUS_RETURN_IF_FAIL(status, KernelPtr());
 
-                if (!localStatus.ok())
-                {
-                    services::internal::tryAssignStatus(status, localStatus);
-                    return KernelPtr();
-                }
                 kernel.reset(new OpenClKernelLevelZero(_executionTarget, *_currentProgramRef, kernelRef));
```

Comment:
Is it impossible to check the status of operation here?

Reply:
This PR contains change only `Status *` -> `Status &`, the other thing will be in different PR